config/ufs/machines/config_machines.xml

```diff
@@ -451,7 +451,77 @@ This allows using a different mpirun command to launch unit tests
       <env name="W3NCO_LIBd">$ENV{NCEPLIBS_DIR}/lib/libw3nco_v2.0.6_d.a</env>
     </environment_variables>
   </machine>
-
+ <machine MACH="gaea">
+    <DESC>
+      port to gaea
+    </DESC>
+    <NODENAME_REGEX>gaea*</NODENAME_REGEX>
+    <OS>CNL</OS>
+    <COMPILERS>cray-intel</COMPILERS>
+    <MPILIBS>mpich</MPILIBS>
+    <PROJECT>esrl</PROJECT>
+    <SAVE_TIMING_DIR> </SAVE_TIMING_DIR>
+    <CIME_OUTPUT_ROOT>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/scratch</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/inputdata</DIN_LOC_ROOT>
+    <DIN_LOC_ROOT_CLMFORC>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/inputdata/lmwg</DIN_LOC_ROOT_CLMFORC>
+    <DOUT_S_ROOT>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/archive/$CASE</DOUT_S_ROOT>
+    <BASELINE_ROOT>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/baselines</BASELINE_ROOT>
+    <CCSM_CPRNC>$ENV{SCRATCH}/$ENV{USER}/ufs-runs/tools/cprnc/cprnc</CCSM_CPRNC>
+    <GMAKE>make</GMAKE>
+    <GMAKE_J>8</GMAKE_J>
+    <BATCH_SYSTEM>slurm</BATCH_SYSTEM>
+    <SUPPORTED_BY>mpotts@redlineperf.com</SUPPORTED_BY>
+    <MAX_TASKS_PER_NODE>16</MAX_TASKS_PER_NODE>
+    <MAX_MPITASKS_PER_NODE>16</MAX_MPITASKS_PER_NODE>
+    <PROJECT_REQUIRED>FALSE</PROJECT_REQUIRED>
+    <mpirun mpilib="default">
+     <executable>srun</executable>
+      <arguments>
+        <arg name="label"> --label</arg>
+        <arg name="num_tasks"> -n {{ total_tasks }}</arg>
+    </arguments>
+    </mpirun>
+    <module_system type="module" allow_error="true">
+      <init_path lang="perl">/opt/cray/pe/modules/3.2.10.5/init/perl.pm</init_path>
+      <init_path lang="python">/opt/cray/pe/modules/3.2.10.5/init/python.py</init_path>
+      <init_path lang="csh">/opt/cray/pe/modules/3.2.10.5/init/csh</init_path>
+      <init_path lang="sh">/opt/cray/pe/modules/3.2.10.5/init/sh</init_path>
+      <cmd_path lang="perl">/opt/cray/pe/modules/4.1.3.1/bin/modulecmd perl</cmd_path>
+      <cmd_path lang="python">/opt/cray/pe/modules/4.1.3.1/bin/modulecmd python</cmd_path>
+      <cmd_path lang="sh">module</cmd_path>
+      <cmd_path lang="csh">module</cmd_path>
+      <modules>
+        <command name="purge"/>
```

Comment:
purge does not work on cray.  Instead I usually have a list of specific modules to remove such as this https://github.com/ESMCI/cime/blob/master/config/cesm/machines/config_machines.xml#L712